.github/workflows/ci.yaml

```diff
@@ -2303,6 +2706,56 @@ jobs:
       - name: ghcr.io/hyperledger/cactus-quorum-multi-party-all-in-one
         run: DOCKER_BUILDKIT=1 docker build ./tools/docker/quorum-multi-party-all-in-one/ -f ./tools/docker/quorum-multi-party-all-in-one/Dockerfile -t cactus-quorum-multi-party-all-in-one
     
+      - name: Run Trivy vulnerability scan for cactus-quorum-multi-party-all-in-one
+        uses: aquasecurity/trivy-action@0.11.2
+        with:
+          image-ref: 'cactus-quorum-multi-party-all-in-one'
+          format: 'table'
+          exit-code: '1'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+          severity: 'CRITICAL,HIGH'
+  generate_coverage_report:
+    runs-on: ubuntu-22.04
+    needs: [check-coverage, cp-bungee-hermes]
+    if: needs.check-coverage.outputs.run-coverage == 'true'
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v4.1.1
+
+    - name: Set up Node.js
+      uses: actions/setup-node@v4.0.2
+      with:
+        node-version: ${{ env.NODEJS_VERSION }}
+
+    - name: Restore Yarn Cache
+      uses: actions/cache@v4.0.1
+      with:
+        path: ./.yarn/
+        key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+
+    - name: Install dependencies and istanbul-merge
+      run: |
+        yarn install --frozen-lockfile
+        yarn add istanbul-merge --dev
+
+    - name: Download coverage reports
+      uses: actions/download-artifact@v4.1.1
+      with:
+        path: ./code-coverage-ts/
+
+    - name: Merge and generate coverage reports
+      run: |
+        ls -R ./code-coverage-ts/**/**/coverage-final.json
+        npx istanbul-merge --out coverage.json ./code-coverage-ts/coverage-reports-*/*/coverage-final.json
+        npx istanbul report --include coverage.json --dir cacti html
+        npx istanbul report --include coverage.json --dir cacti text
+        npx istanbul report --include coverage.json --dir cacti text-summary
+
```

Comment:
@aldousalvarez I see that this generates the final, combined report, but does it also get uploaded to [CodeCov](https://about.codecov.io/) which would then propgate to the codecov badge on the root README.md file?

I thought we'd need something like this as the final step to make sure that the code coverage file gets uploaded (and also to be able to verify that it's in the valid format accepted by codecov):

https://github.com/codecov/codecov-action

Reply:
@petermetz okay got it, I will work on this one and let you know as soon as it is done. Thank you

Reply:
@aldousalvarez Thank you, this part of it looks good now.